base/reduce.jl

```diff
@@ -396,7 +396,11 @@ end
 
 
 ## prod
+"""
+    prod(f, iter)
```

Comment:
Yeah, and adjust https://github.com/JuliaLang/julia/blob/master/doc/src/stdlib/collections.md#iterable-collections to split into 2 entries. I'm not sure what the most general signature is for product of an array over a dimension, but it's probably not `prod(::Any, ::Any)` ?